components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -1238,18 +1238,24 @@ public static String getApplicationUpdatedVersion(ServiceProvider serviceProvide
         String currentVersion = serviceProvider.getApplicationVersion();
         String inboundConfigType = getInboundConfigType(serviceProvider);
 
-        // Since there will be new versions onboarded in the future, initialized this as a switch not if.
-        if (currentVersion.equals(ApplicationConstants.ApplicationVersion.APP_VERSION_V0)) {
-            if (!inboundConfigType.equals(IdentityApplicationConstants.OAuth2.NAME)) {
-                currentVersion = ApplicationConstants.ApplicationVersion.APP_VERSION_V1;
-            }
+        switch (currentVersion) {
+            case ApplicationConstants.ApplicationVersion.APP_VERSION_V0:
+            case ApplicationConstants.ApplicationVersion.APP_VERSION_V1:
+                if (inboundConfigType != null && !inboundConfigType.equals(IdentityApplicationConstants.OAuth2.NAME)) {
+                    currentVersion = ApplicationConstants.ApplicationVersion.APP_VERSION_V2;
```

Comment:
Is this logic correct? If someone has already upgraded his application to V1 after modifying the application logic, are we automatically upgrading the version of those applications to v2?

Reply:
No. The logic here is, if the app version is either v0 or v1 AND the app inbound auth type is not OIDC then automatically upgrade the version to v2.